### PR TITLE
#162414278 Implement in-app and email notifications

### PR DESCRIPTION
**Pull request by jesseinit**
#### What does this PR do?

- Implement in-app notifications and the sending of email when a user gets followed, comments on an article and when a new article is created.

#### Description of Task to be completed?

- Install required packages
- Implement events to aid seamless triggering of events
- Write tests

#### How should this be manually tested?

- Clone the repo `git clone https://github.com/andela/neon-ah-backend.git`
- Install Postgresql from [here](https://www.postgresql.org/download/)
- Create a database `haven_dev`
- Checkout to the branch `git checkout -b ft-inapp-email-notification-162414278`
- Install application dependencies `npm install`
- Run the tests `npm test`

#### What are the relevant screenshots?
screenshot 2019-01-25 at 4 35 00 pm
screenshot 2019-01-25 at 4 35 35 pm
screenshot 2019-01-25 at 4 35 50 pm

#### What are the relevant pivotal tracker stories?

[In-App and Email Notifications](https://www.pivotaltracker.com/story/show/162414278)

### CODE REVIEW

**Comment by Makwe-O** [comment]
I like the little details you added with the links and the emoji. Impressive work. I feel the new article alert should show the username. 

LGTM

**Review by rafmme — COMMENTED** [review]
%

File: server/helpers/emailTemplates/newFollowerTemplate.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    </style>
+  </head>
+  <body>
+    <div class="container">
+      <img class="logo"
+        src="https://res.cloudinary.com/jesseinit/image/upload/v1547303525/Logo.png"
+        alt="logo"
+      />
+      <h3 class="username">Hi ${user.fullName},</h3>
+      <p class="message">Huraayy 🚀🚀 <a href=http://localhost:3000/users/${followerUserName}>${followerUserName}</a> just followed you.
```
Comment (rafmme) [inline server/helpers/emailTemplates/newFollowerTemplate.js:24]: Nice work Jesse but I'll like to raise a concern. The URL uses localhost which seems static, How will it be updated when deployed to production (Heroku)?

**Comment by tejiri4** [comment]
LGTM, yet.

**Comment by ebzeal** [comment]
LGTM. I like that you have a nicely laid out PR summary

File: server/helpers/emailTemplates/newFollowerTemplate.js (hunk 1 above)
Comment (jesseinit) [reply server/helpers/emailTemplates/newFollowerTemplate.js:24]: We are yet to have a front-end url. When we have that I'll be updated to facilitate site redirect.

**Comment by tejiri4** [comment]
User notification on follow is not working.

**Review by jesseinit — COMMENTED** [review]
Looks good to me. 👍🏾

**Pull request merged**